pom.xml

```diff
@@ -55,6 +55,11 @@
   </pluginRepositories>
 
   <dependencies>
+    <dependency>
```

Comment:
NIT: I wonder if it would be better to depend on the gitlab-api plugin and get the gitlab4j library transitively so you are using (and in a limited way also testing) the library the gitlab-branch-source plugin is using internally.

Reply:
I would say not

the library may be updated to work with newer GitLab servers (which we update here) where it is not updated in the plugin and as such we would want to know the plugin is broken when GL is updated rather than the test errors for some issue that is more obscure.

There is no requirement to use the same version of the library or even the same library when interacting with any system.

Reply:
If I remove the `gitlab4j` dependency and add `gitlab-api` there are several issues. Is it worth updating these dependencies?
```[ERROR] Rule 4: org.apache.maven.enforcer.rules.dependency.RequireUpperBoundDeps failed with message:
[ERROR] Failed while enforcing RequireUpperBoundDeps. The error(s) are [
[ERROR] Require upper bound dependencies error for org.javassist:javassist:3.27.0-GA paths to dependency are:
[ERROR] +-org.jenkins-ci:acceptance-test-harness:999999-SNAPSHOT
[ERROR]   +-com.browserup:browserup-proxy-core:2.1.2
[ERROR]     +-org.javassist:javassist:3.27.0-GA
[ERROR] and
[ERROR] +-org.jenkins-ci:acceptance-test-harness:999999-SNAPSHOT
[ERROR]   +-io.jenkins.plugins:gitlab-api:5.3.0-91.v1f9a_fda_d654f
[ERROR]     +-io.jenkins.plugins:jersey2-api:2.39.1-1
[ERROR]       +-org.glassfish.jersey.inject:jersey-hk2:2.39.1
[ERROR]         +-org.javassist:javassist:3.29.0-GA

Reply:
just don't depend on the plugin and keep it as is.

Reply:
ah, sorry, I didn't refresh the page and I didn't see your previous comment before posting mine.